internal/utils/set_test.go

```diff
@@ -0,0 +1,81 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewSet(t *testing.T) {
+	set := NewSet[int]()
+	assert.NotNil(t, set)
+	assert.Equal(t, 0, len(set))
```

Comment:
This would be better maybe

```suggestion
	assert.Empty(t, set)
```